.github/workflows/release.yaml

```
@@ -0,0 +1,162 @@
+        with:
+          path: "*"
+          key: ${{ runner.os }}-build-${{ env.cache-name }}-${{ hashFiles('*')}}
+          restore-keys: |
+            ${{ runner.os }}-build-${{ env.cache-name }}-
+      - name: Debug LS
+        run: |
+          pwd
+          ls -alh
+      - name: Node Lint
+        run: npm run lint
+      - name: NPM Build
+        run: npm run build
+      - name: Cache Build Assets
+        uses: actions/cache@v2
+        env:
+          cache-name: cache-build-dist
+        with:
+          path: dist
+          key: ${{ runner.os }}-build-${{ env.cache-name }}-${{ hashFiles('webpack.config.js', 'dist/**')}}
+  release:
+    needs: build
+    name: Release
+    runs-on: ubuntu-latest
+    steps:
+      - name: Retrieve Build Assets
+        uses: actions/cache@v2
+        env:
+          cache-name: cache-build-dist
+        with:
+          path: dist
+          key: ${{ runner.os }}-build-${{ env.cache-name }}-${{ hashFiles('webpack.config.js', 'dist/**')}}
+          restore-keys: |
+            ${{ runner.os }}-build-${{ env.cache-name }}-
+      - name: Get Next Version Number
+        id: next-version
+        uses: actions/github-script@v3.1.0
+        with:
+          github-token: ${{ secrets.GITHUB_TOKEN }}
+          script: |
+            var currentReleaseName = ""
+            try {
+              const getReleaseResp = await github.repos.getLatestRelease({
+                owner: context.repo.owner,
+                repo: context.repo.repo
+              })
+              currentReleaseName = getReleaseResp.data.name
+              console.log(`Current Release Name: ${currentReleaseName}`)
+            } catch(err) {
+              if (err.name == 'HttpError') {
+                console.warn("Found HttpError")
+                if (err.status == 404)  {
+                  console.warn("Error 404. No latest release found. Starting from 0.")
+                  currentReleaseName = "v0"
```

Honestly I would throw an error here instead of restarting from zero. Unless there's a realistic scenario where we'd want to use this? We're automating a lot of the release side, and I can visualise a hidden mistake happening somewhere and our release numbering suddenly restarting from 0, and days going by without us realising...